quinn-proto/src/tests/util.rs

```diff
@@ -310,26 +310,25 @@ impl TestEndpoint {
 
         while self.inbound.front().map_or(false, |x| x.0 <= now) {
             let (recv_time, ecn, packet) = self.inbound.pop_front().unwrap();
-            if let Some((ch, event)) = self.endpoint.handle(recv_time, remote, None, ecn, packet) {
+            if let Some(event) = self.endpoint.handle(recv_time, remote, None, ecn, packet) {
                 match event {
-                    DatagramEvent::NewConnection(conn) => {
+                    DatagramEvent::NewConnection(ch, conn) => {
                         self.connections.insert(ch, conn);
                         self.accepted = Some(ch);
                     }
-                    DatagramEvent::ConnectionEvent(event) => {
+                    DatagramEvent::ConnectionEvent(ch, event) => {
                         self.conn_events
                             .entry(ch)
                             .or_insert_with(VecDeque::new)
                             .push_back(event);
                     }
+                    DatagramEvent::Response(transmit) => {
+                        self.outbound.extend(split_transmit(transmit));
+                    }
                 }
             }
         }
 
-        while let Some(x) = self.poll_transmit() {
```

Comment:
I suppose this changes the order of things. Have you reasoned about why it's okay to change things? Wasn't there a reason to do "metadata" events before I/O?

Reply:
I don't think this changes any behavior, because appending to `self.outbound` has no side effects.

Reply:
Ah, fair. So what about for other downstream users of the quinn-proto crate?

Reply:
Transmits not associated with any connection are by definition not associated with any persistent state, so no special care should be required, even leaving aside the fact that we could never rely on the ordering of real I/O to begin with.